From: Debajyoti Bera <dbera.web@gmail.com>

Bim entry with marker "PHUT" is not handled by Bim.cs.

Index: JpegHeader.cs
===================================================================
--- JpegHeader.cs	(revision 3340)
+++ JpegHeader.cs	(working copy)
@@ -388,7 +388,14 @@
 			using (System.IO.Stream bimstream = new System.IO.MemoryStream (marker.Data, len,
 											marker.Data.Length - len, false)) {
 
-				Bim.BimFile bim = new Bim.BimFile (bimstream);
+				Bim.BimFile bim;
+				try {
+					bim = new Bim.BimFile (bimstream);
+				} catch {
+					// Bim entry with marker "PHUT" is not handled by Bim.cs
+					return null;
+				}
+
 				// FIXME: What about EntryType.XMP ?
 				Bim.Entry iptc_entry = bim.FindEntry (Bim.EntryType.IPTCNAA);
 				if (iptc_entry == null)
